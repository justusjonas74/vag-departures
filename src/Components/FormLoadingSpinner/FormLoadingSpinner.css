#loadingSpinner {
    position: absolute;
    margin-left: -40px;
    z-index: 99999;
    line-height: 2.3rem;
}
