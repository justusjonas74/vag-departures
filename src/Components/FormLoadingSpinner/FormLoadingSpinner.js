import React, { Component } from 'react';
import './FormLoadingSpinner.css';

class FormLoadingSpinner extends Component {
    render() {
        const isLoading = this.props.isLoading;
        if (isLoading) {
            return (
                <i className="fa fa-spinner fa-pulse" id="loadingSpinner"></i>
            )
        }
        else {
            return null
        }
    }
}

export default FormLoadingSpinner