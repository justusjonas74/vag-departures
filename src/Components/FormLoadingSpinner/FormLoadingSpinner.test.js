import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';


import FormLoadingSpinner from './FormLoadingSpinner';



describe('FormLoadingSpinner', () => {
  // UNIT TESTS 
  it('should render correctly', () => {
    shallow(<FormLoadingSpinner />);
  });
  
  it('renders correctly', () => {
    const tree = renderer
      .create(<FormLoadingSpinner />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});


// SNAPSHOT TESTS

