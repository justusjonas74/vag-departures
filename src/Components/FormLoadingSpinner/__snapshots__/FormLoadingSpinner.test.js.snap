// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLoadingSpinner renders correctly 1`] = `
<i
  className="fa fa-spinner fa-pulse"
  id="loadingSpinner"
/>
`;
